apps/crm/crm/src/app/crm/pages/external-sales/view/view.component.ts

```diff
@@ -0,0 +1,96 @@
+import { Component, ChangeDetectionStrategy, Pipe, PipeTransform } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { MatDialog } from '@angular/material/dialog';
+import { FormControl } from '@angular/forms';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { RouterQuery } from '@datorama/akita-ng-router-store';
+import { appName, getCurrentApp } from '@blockframes/utils/apps';
+import { OrganizationService } from '@blockframes/organization/+state';
+import { Sale, ContractService, Holdback } from '@blockframes/contract/contract/+state';
+import { MovieService } from '@blockframes/movie/+state';
+import { joinWith } from '@blockframes/utils/operators';
+import { getSeller } from '@blockframes/contract/contract/+state/utils'
+import { of } from 'rxjs';
+import { filter, map, switchMap, tap } from 'rxjs/operators';
+import { IncomeService } from '@blockframes/contract/income/+state';
+import { Term } from '@blockframes/contract/term/+state';
+import { ConfirmInputComponent } from '@blockframes/ui/confirm-input/confirm-input.component';
+import { Negotiation } from '@blockframes/contract/negotiation/+state/negotiation.firestore';
+import { isInitial } from '@blockframes/contract/negotiation/utils';
+
+@Component({
+  selector: 'contract-view',
+  templateUrl: './view.component.html',
+  styleUrls: ['./view.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class ContractViewComponent {
+  public app = getCurrentApp(this.routerQuery);
+  public appName = appName[this.app];
+  public orgId = this.orgService.org.id;
+
+
+  contract$ = this.route.params.pipe(map(r => r.saleId as string))
+    .pipe(
+      switchMap(saleId => this.getSale(saleId)),
+      filter(contract => !!contract),
+      tap(contract => {
+        this.statusForm.setValue(contract.status);
+      })
```

Comment:
I think you can inline it.
```suggestion
      tap(contract => this.statusForm.setValue(contract.status))
```